apps/website/src/components/LeftSidebarContent.astro

```diff
@@ -1,109 +1,86 @@
 ---
+import { getCollection } from 'astro:content';
+import { html } from 'htm/react';
+
 const currentPage = Astro.url.pathname.endsWith('/')
   ? Astro.url.pathname.slice(0, -1)
   : Astro.url.pathname;
 
-const docsPages = (await Astro.glob('../pages/docs/*.mdx')).map(({ url, frontmatter }) => ({
-  url,
-  title: frontmatter.title,
-  group: frontmatter.group,
-}));
+const introPages = await getCollection('docs', ({ data: { group } }) => group === 'intro');
+const mainComponents = await getCollection('docs', ({ data: { group } }) => group === undefined);
+const utilityComponents = await getCollection(
+  'docs',
+  ({ data: { group } }) => group === 'utilities',
+);
 
-const introPages = docsPages.filter((page) => !page.group);
+const NavLink = ({ slug = '', title = '', ...props }) => {
+  const url = `/docs/${slug}`;
+  const isCurrent = currentPage === url;
 
-const sections = new Map<string, typeof docsPages>();
-sections.set(
-  'core',
-  docsPages.filter((item) => item.group === 'core'),
-);
-sections.set(
-  'buttons',
-  docsPages.filter((item) => item.group === 'buttons'),
-);
-sections.set(
-  'inputs',
-  docsPages.filter((item) => item.group === 'inputs'),
-);
-sections.set(
-  'typography',
-  docsPages.filter((item) => item.group === 'typography'),
-);
-sections.set(
-  'utilities',
-  docsPages.filter((item) => item.group === 'utilities'),
-);
+  return html`<a href="${url}" aria-current=${isCurrent ? 'page' : 'false'} ...${props}>
```

Comment:
To confirm, what "repition" was reduced after using `htm`?

Reply:
without htm, i would need to repeat all this code three times (each instance of NavLink)
![image](https://github.com/iTwin/iTwinUI/assets/9084735/9b47c8b5-1fc4-4f66-a3d5-03ab4689092a)

basically, htm allows me to define inline components in the frontmatter